api/src/pcapi/core/search/__init__.py

```diff
@@ -381,6 +385,46 @@ def index_offers_of_venues_in_queue() -> None:
         logger.exception("Could not index offers of venues from queue")
 
 
+def get_base_query_for_offer_indexation() -> BaseQuery:
+    return (
+        offers_models.Offer.query.options(
+            sa.orm.joinedload(offers_models.Offer.venue).joinedload(offerers_models.Venue.managingOfferer)
+        )
+        .options(sa.orm.joinedload(offers_models.Offer.criteria))
+        .options(sa.orm.joinedload(offers_models.Offer.mediations))
+        .options(sa.orm.joinedload(offers_models.Offer.product))
+        .options(sa.orm.joinedload(offers_models.Offer.stocks))
+    )
+
+
+BASE_QUERY_FOR_LAST_30_DAYS_BOOKINGS = (
+    sa.select(
+        offers_models.Offer.id.label("offer_id"), sa.func.count(bookings_models.Booking.id).label("bookings_number")
+    )
+    .select_from(offers_models.Offer)
+    .outerjoin(offers_models.Stock, offers_models.Stock.offerId == offers_models.Offer.id)
+    .outerjoin(bookings_models.Booking, bookings_models.Booking.stockId == offers_models.Stock.id)
+    .where(
+        sa.or_(
+            bookings_models.Booking.dateCreated >= datetime.datetime.utcnow() - datetime.timedelta(days=30),
+            bookings_models.Booking.dateCreated.is_(None),
+        ),
+        sa.or_(
+            sa.not_(
+                bookings_models.Booking.status.in_(
+                    (
+                        bookings_models.BookingStatus.PENDING,
+                        bookings_models.BookingStatus.CANCELLED,
+                    )
+                )
+            ),
+            bookings_models.Booking.status.is_(None),
+        ),
```

Comment:
(détail) `Booking.status` est `NOT NULL`.

```suggestion
        bookings_models.Booking.status.notin_(
            (
                bookings_models.BookingStatus.PENDING,
                bookings_models.BookingStatus.CANCELLED,
            )
        )
    ),
```

Et on peut utiliser `BookingStatus.notin_(...)` qui me semble plus clair que `not_(Booking.status.in_(...))`.